processing/src/main/java/io/druid/query/topn/BaseTopNAlgorithm.java

```diff
@@ -101,6 +104,8 @@ public void run(
 
   protected abstract DimValSelector makeDimValSelector(Parameters params, int numProcessed, int numToProcess);
 
+  protected abstract int computeNewLength(DimValSelector dimValSelector, int numProcessed, int numToProcess);
```

Comment:
instead of making the method abstract, maybe we can return numToProcess by default, since that seems to be the case for most implementations?